src/org/opendatakit/briefcase/model/BriefcasePreferences.java

```diff
@@ -77,17 +79,44 @@ public String get(String key, String defaultValue) {
         return preferences.get(key, defaultValue);
     }
 
+    /**
+     * Returns the value associated with the specified key in this preference
+     * node mapped by a function that takes a String and returns a value of type T.
+     * Returns the specified default if there is no value associated with
+     * the key, or the backing store is inaccessible.
+     *
+     * @param key
+     *          key whose associated value is to be returned.
+     * @param mapper
+     *          function that takes a String and returns a value of type T
+     * @param defaultValue
+     *          the value to be returned in the event that this preference node
+     *          has no value associated with key.
+     * @param <T>
+     *          type of the output of this method
+     * @return the value associated with key, or defaultValue if no value is associated
+     *         with key, mapped with the mapper function, or the backing store is inaccessible.
+     */
+    public <T> T get(String key, Function<String,T> mapper, T defaultValue) {
+      return Optional.ofNullable(preferences.get(key, null)).map(mapper).orElse(defaultValue);
+    }
+
     /**
      * Associates the specified value with the specified key in this preference
      * node.
      *
+     * If the value is null, then the key is removed
+     *
      * @param key
      *          key with which the specified value is to be associated.
      * @param value
-     *          value to be associated with the specified key.
+     *          value to be associated with the specified key or null.
      */
     public void put(String key, String value) {
-        preferences.put(key, value);
+        if (value != null)
+            preferences.put(key, value);
+        else
+            remove(key);
```

Comment:
I haven't seen this type of usage before where `put(key, null)` removes a key. Is this typical?

Reply:
It's just a convenient API for users of `BriefcasePreferences`. Take a look at how this is used in [line 148 of ExportPanel](https://github.com/opendatakit/briefcase/pull/264/commits/108397f99047f15fea7577f224b464bdafa9b75f#diff-53cd7276f9b5fe0579443fdf5f255e79R148).

The fact that the key gets removed from the Preferences is just an implementation detail that users of this class don't have to know. Also, removing the key is just what I'd have to do in [the same line](https://github.com/opendatakit/briefcase/pull/264/commits/108397f99047f15fea7577f224b464bdafa9b75f#diff-53cd7276f9b5fe0579443fdf5f255e79R148) inside an ugly if block inside the listener function.

The next step would be to accept `Optional<String>` values in `BriefcasePreferences.put(...)` to avoid handling nulls and simplifying even more the code at use site.